src/vivarium/interpolation.py

```diff
@@ -68,7 +68,11 @@ def __init__(
         if self.key_columns:
             # Since there are key_columns we need to group the table by those
             # columns to get the sub-tables to fit
-            sub_tables = self.data.groupby(list(self.key_columns))
+            if len(self.key_columns) == 1:
+                # Pass element instead of list of length 1 to avoid FutureWarning
+                sub_tables = self.data.groupby(self.key_columns[0])
+            else:
+                sub_tables = self.data.groupby(list(self.key_columns))
```

Comment:
I would just pass self.key_columns as a pd.Series like you did in the state_machine module or just be consistent with how you are handling this.